src/observer.js

```diff
@@ -1,11 +1,11 @@
 import React, { Component, PureComponent } from "react"
 import hoistStatics from "hoist-non-react-statics"
-import { createAtom, Reaction, _allowStateChanges, $mobx } from "mobx"
+import * as mobx from "mobx"
```

Comment:
The problem is that imports like these kill tree-shaking (although that mobx is not the most tree shakeable lib). I'll keep this PR open for a bit, it might be good stop-gap if more people run into the same problem and try to combine mobx-react 5.2+ with mobx 5.3.1-